wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -93,13 +93,15 @@ def load_path(
 
         extra_args = {}
         if version:
-            obj = self._s3.ObjectVersion(bucket, key, version).Object()
+            obj_version = self._s3.ObjectVersion(bucket, key, version)
             extra_args["VersionId"] = version
+            etag = self._etag_from_obj_version(obj_version)
+            obj = obj_version.Object()
```

Comment:
we know this code is buggy, what do we use `obj` for later? This `obj` is always the latest and not the versioned object so getting it feels dangerous.

Reply:
we need the object instance to download the file. we're recording the version_id in the extra_args dictionary, which is passed while downloading the object at the bottom of the method in `obj.download_fileobj(f, ExtraArgs=extra_args)`

Reply:
I see, have you confirmed we actually download the right version and not the latest?

Reply:
yep!